### PR TITLE
Modifying test harness for 10x test development velocity.  Client now…

### PR DESCRIPTION
**Pull request by bryanmau1**
… uploads new test files to cloud storage per test and the test app automatically pulls them down for testing.  This bypasses deployment.

R: @jonparrott 
CC: @andrewsg @jcollins-g

### CODE REVIEW

File: tests/e2e/app/app.yaml [hunk 1]
```diff
@@ -4,3 +4,6 @@ vm: true
 handlers:
```
Comment (jonparrott) [inline tests/e2e/app/app.yaml]: If this is generated, delete it and add it to .gitignore.

File: tests/e2e/app/main.py [hunk 2]
```diff
@@ -32,7 +38,19 @@ def capture():
...
     def get(self):
+        # Set up test directory.  Done once per app upload.
+        if not os.path.isdir('tests'):
+          os.mkdir('tests')
+
+        # Refresh the tests from cloud storage.
+        bucket = storage.Client().get_bucket(GCLOUD_STORAGE_BUCKET)
+        blob_iter = bucket.list_blobs()
+        for blob in blob_iter:
+          blob.download_to_filename('tests/%s' % blob.name)
```
Comment (jonparrott) [inline tests/e2e/app/main.py]: How does this deal with the case of deleted files?

File: tests/e2e/common.mk [hunk 3]
```diff
@@ -0,0 +1 @@
+BUCKET=gs://${GCLOUD_PROJECT}-python-compat-e2e-tests
```
Comment (jonparrott) [inline tests/e2e/common.mk]: Just include this in the top-level `common.mk`.

File: tests/e2e/app/main.py (hunk 2 above)
Comment (bryanmau1) [inline tests/e2e/app/main.py]: It doesn't.  Good point :)  I'll wipe the directory each time.  It takes minimal time to copy anyways especially compared to redeploying.

**Comment by bryanmau1** [comment]
Updated with comments fixed.

File: tests/e2e/app/main.py (hunk 2 above)
Comment (jonparrott) [inline tests/e2e/app/main.py]: Seems you dealt with it on the upload but not here on the download. `shutil.rmtree`?

**Comment by jonparrott** [comment]
LGTM with final comment.

File: tests/e2e/app/main.py (hunk 2 above)
Comment (bryanmau1) [inline tests/e2e/app/main.py]: On upload, we use gsutil rm bucket/** to delete everything in cloud storage.
On download, we first clear the tests directory with shutil.rmtree which is equivalent to 'rm -rf'

Comment (jonparrott) [inline tests/e2e/app/main.py]: Oh I totally missed that. Good deal.

**Pull request merged**